src/BloomBrowserUI/bookEdit/js/editViewFrame.js

```diff
@@ -137,10 +110,154 @@ function initializeReaderSetupDialog() {
+        else if ((sw < 723) || (window.innerHeight < 583)) {
+            h = 460;
+            w = 580;
+        }
+
+        theDialog = $(dialogContents).dialog({
+            autoOpen: false,
+            modal: true,
+            width: w,
+            height: h,
+            position: {
+                my: "left top", at: "left top", of: window
+            },
+            title: title,
+            buttons: {
+                OK: {
+                    text: addButtonText,
+                    icons: {
+                        primary: "ui-icon-plusthick"
+                    },
+                    click: function () {
+                        document.getElementById('addPage_frame').contentWindow.postMessage('AddSelectedPage', '*');
+                    }
+                },
+                Cancel: {
+                    text: parentElement.localizationManager.getText('Common.Cancel', 'Cancel'),
+                    click: function () {
+                        $(this).dialog("close");
+                    }
+                }
+            },
+            close: function() {
+                $(this).remove();
+                fireCSharpEvent('setModalStateEvent', 'false');
+            },
+            open: function () {
+                adjustAddPageButton(addButtonText);
+                setTimeout(function() {
+                    setDialogInnerIframeWidth();
+                }, 200);
+            }
+        });
+        fireCSharpEvent('setModalStateEvent', 'true');
+        theDialog.dialog('open');
+    });
+}
+
+function adjustAddPageButton(addButtonText) {
+    var button = $('.ui-dialog-buttonpane').find('button:contains('+addButtonText+')');
+    button.width(button.width() + 15);
+}
+
+function setDialogInnerIframeWidth() {
+    var $this = $('#addPage_frame');
+    var width = $this.contents().find('#mainContainer').width();
+    var height = $this.contents().find('#mainContainer').height();
+    $this.height(height);
+    $this.width(width);
+}
+
+function localizeDialogContents(dialogContents, description, blankMessage) {
+    $(dialogContents).find('iframe').contents().find('.DescriptionHeader').text(description);
+    $(dialogContents).find('iframe').contents().find('iframe').contents().find('#innerBox').text(blankMessage);
+}
+
+function getAddPageDialogLocalizedStrings() {
+    // Without preloading these, they are not available when the dialog is created
+    var pairs = {};
+    pairs['AddPageDialog.Title'] = 'Add Page...';
+    pairs['AddPageDialog.DescriptionLabel'] = 'Description';
+    pairs['AddPageDialog.PreviewMessage'] = 'This will contain a preview of a template page when one is selected.';
+    //pairs['HelpMenu.Help Menu'] = 'Help';
+    pairs['AddPageDialog.AddPageButton'] = 'Add This Page';
+    pairs['Common.Cancel'] = 'Cancel';
+    return pairs;
+}
+
+//noinspection JSUnusedGlobalSymbols
+// Used by the addPage_frame to initialize the setup dialog
+function initializeAddPageDialog(templatesJSON) {
```

Comment:
I think here it might be helpful to give an example of a typical template that might be passed in and/or explain what is supposed to be initialized by what JSON.